onnxscript/rewriter/generic_pattern.py

```diff
@@ -1093,136 +844,86 @@ def make_rule(self) -> orp.RewriteRule:
         return GenericRewriteRule(self)
 
 
-class OnnxGenericPattern(GenericPattern):
-    """An instance of GenericPattern taking onnx model.
+class FunctionPattern(GenericPattern):
+    """An instance of GenericPattern taking ir.Function.
 
     It defines the matching pattern and its replacement.
 
-    :param match_proto: the onnx function defining the matching pattern
-    :param apply_proto: the onnx function defining the new pattern
-    :param validate_mapping: the function used to validate a pattern
-    :param use_onnxscript: tells if the apply_proto is an onnxscript function or
-        a regular function returning a FunctionProto
-    :param opsets: opset to consider when converting the function into ONNX,
-        if not specified, it is opset 18 for the main opset, and opset 1
-        for domain com.microsoft.
-    :param verbose: in [0, 10], increase the verbosity to understand why a pattern
-        does not match
+    Args:
+        match_pattern: the onnx ir function defining the matching pattern
+        apply_pattern: the onnx ir function defining the new pattern
+        validate_mapping: the function used to validate a pattern
+        verbose: in [0, 10], increase the verbosity to understand why a pattern
+            does not match
+
     """
 
     def __init__(
         self,
-        match_proto: onnx.FunctionProto,
-        apply_proto: onnx.FunctionProto | typing.Callable,
-        validate_mapping: typing.Callable,
-        use_onnxscript: bool = True,
-        opsets: dict[str, onnxscript.values.Opset] | None = None,
+        match_pattern: ir.Function,
+        apply_pattern: ir.Function,
+        validate_mapping,
         verbose: int = 0,
     ):
-        super().__init__(verbose=verbose)
-        self.match_proto = match_proto
-        self._validate_mapping = validate_mapping
-        self.apply_proto = apply_proto
-        self.use_onnxscript = use_onnxscript
-        self.opsets = opsets
-        self._cache = {}
-
-    def validate_mapping(self, g: oir.Model, match_result: PatternMatchResult) -> bool:
-        """Evaluates the consistency of the replacements."""
-        return self._validate_mapping(g, match_result)
+        self.match_pattern = match_pattern
+        self.apply_pattern = apply_pattern
+        self.validate_mapping = validate_mapping
+        self.verbose = verbose
 
-    def _build_pattern(
```

Comment:
This function should not have been removed. It creates the replacements nodes based on the parameters obtained after the matching.